### PR TITLE
ci: add certificate auto-update workflow

### PR DESCRIPTION
**Pull request by stuckinforloop**
### Description

Add workflow file to check and auto update certificates. If a new certificate is found, we regenerate `certificates.ml` and `pem.ml` files and create a new patch release.

### CODE REVIEW

File: regen.ml [hunk 1]
```diff
@@ -85,7 +85,7 @@ let generate_cas pem =
     String.concat ";\n" (List.map (fun ca -> Format.asprintf "{|%s|}" ca) cas)
   in
   let file = Format.sprintf "let certificates = [\n%s\n] " cas in
-  let oc = open_out "cas.ml" in
+  let oc = open_out "certificates.ml" in
```
Comment (stuckinforloop) [inline regen.ml:88]: While testing regen locally, I found a bug where we are creating a new `cas.ml` file with the same content as `certificates.ml` file. If this is desired, I'll revert it back.

Comment (leostera) [reply regen.ml:88]: This looks good, it was likely an oversight!

File: .github/workflows/cert.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Bump version and create release
+      if: steps.check_changes.outcome == 'success'
+      run: |
+        current_version=$(git describe --tags --abbrev=0)
+        IFS='.' read -r major minor patch <<< "$current_version"
+        new_version="$major.$minor.$((patch + 1))"
+        
+        git tag "$new_version"
+        git push origin "$new_version"
+        gh release create "$new_version" --generate-notes
```
Comment (leostera) [inline .github/workflows/cert.yml:51]: **question**. Could we change this to `dune-release` to automatically open a PR to Opam? 

`dune-release` should also take care of creating and pushing the git tag to the repo.

Also we'll have to install it with `opam install dune-release`

**Review by leostera — APPROVED** [review]
This looks awesome! ✨ Thanks for the PR @stuckinforloop. Left you a small question about publishing directly to Opam using dune-release, but its not blocking, so I'll merge this! 🙏🏼

**Pull request merged**